### DISCUSSION TITLE
Game ideas

### DISCUSSION THREAD

**Discussion by Anonymous4045** [letsbuilda/pyweek36#9] [Ideas]
The theme is "Dark matter", and it's up to us to interpret this and form it into a game. Here is where we can bounce around ideas until we have a general vision of what we will make.

**Comment by Anonymous4045** [comment]
Since no one else is throwing out ideas, ig I'll say what comes to mind. I don't really understand actual dark matter, and I don't think scientists do either. One concept of it that I think would be cool is light is something you can see, dark is something you can't. If we extend that to matter, then normal matter is something you feel, and dark matter would be something you can't. Maybe this dark matter spreads to normal matter to make it disappear. The player could try to run away from the disappearing floor, sort of a temple run type game.

**Reply by dluiscosta** [reply]
I like that, but would like to add a twist. The character could have a device which allows him to interact with dark matter, but it went crazy and so it keeps going on and off by itself (we could start with a fixed periodicity). This way, dark matter blocks can serve both as "disappearing floors" that the player has to beware of not to fall through, as well as appearing walls that might prevent movement.

**Reply by dluiscosta** [reply]
The game's map would be a side view 2d platform (like the prototype you guys have got from the prior PyWeek) with 3 different blocks:

1- empty block, the player can go through
2 - filled block, the player collides with it and can walk over it
3 - swaps between the prior 2 states periodically

**Reply by dluiscosta** [reply]
This wouldn't necessarily include the concept of spread though...

**Reply by dluiscosta** [reply]
But it still could. If we have spread coming from the bottom of the map, staying there for too long would mean the player falls through everything when the device goes off

**Reply by dluiscosta** [reply]
Instead of a periodic timed thing, we can also make it predicted by a side bar (like a music game) or controlled by the player with a charge (if on for to long, it forceably goes off)

**Reply by dluiscosta** [reply]
I feel like all of these options have a common base implementation which we could start on if enough ppl dig the idea

**Reply by Anonymous4045** [reply]
I think I like the idea of the player being able to control some of the blocks. Maybe there's some evil entity that's turning the floor into dark matter to try to make the player fall, but the player can use a device to turn some walls or platforms to/from dark matter in order to escape. I'd like to hear other people's thoughts about whether this makes sense/would be fun.

**Reply by rfdzan** [reply]
I read this on my search yesterday. it seems roughly similar to what you're describing. it involves ray casting and line of sight, and idk if those  are going to take a long time to implement or not
http://www.squidi.net/three/entry.php?id=86

**Reply by salty-tree** [reply]
I like the idea of dark matter naturally spreading through the level, I think it can go well with using some sort of device (maybe a gun that you can point+shoot with mouse) to temporarily turn platforms/walls back into normal matter
